### PR TITLE
[ecommerce] Using Shared Partials

### PR DESCRIPTION
**Pull request by chrisdanek**

### CODE REVIEW

File: marketplace_builder/views/pages/use-cases/e-commerce/using-shared-partials.liquid [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+    {% if _disabled %} disabled {% endif %}
+  />
+
+  {% include 'forms/error' with _error %}
+</div>
+
+{% endraw %}
+```
+
+#### marketplace_builder/views/partials/forms/error.html
```
Comment (pavelloz) [inline marketplace_builder/views/pages/use-cases/e-commerce/using-shared-partials.liquid]:
```suggestion
#### marketplace_builder/views/partials/forms/error.liquid
```

File: marketplace_builder/views/pages/use-cases/e-commerce/using-shared-partials.liquid [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+  Required params:
+    error: string
+{% endcomment %}
+{%- if error != blank -%}
+  <div class="invalid-feedback">{{ error }}</div>
+{%- endif -%}
+{% endraw %}
+```
+
+#### marketplace_builder/views/partials/forms/label.html
```
Comment (pavelloz) [inline marketplace_builder/views/pages/use-cases/e-commerce/using-shared-partials.liquid]:
```suggestion
#### marketplace_builder/views/partials/forms/label.liquid
```

**Comment by pavelloz** [comment]
This looks like a good candidate to be a tutorial on best practices on its own, referenced from other places - one of them being e-commerce

**Comment by diana-lakatos** [comment]
@pavelloz I think that the topic (how to use partials) should be explained in a beginner tutorial, but for that a simpler example would work better.

File: marketplace_builder/views/pages/use-cases/e-commerce/using-shared-partials.liquid [hunk 3]
```diff
@@ -287,21 +287,21 @@ Let’s take a look at how you could structure the shared partial, to make it co
 {% endraw %}
 ```
 
-This example can be split into three separate parts and as a pattern reused throughout the application
+This example can be split into three separate parts and as a pattern reused throughout the application:
 
-1. Top comment contains a list of all required and optional parameters that can be passed and will be used in this partial. It’s easier this way to see at a glance what options are available
-2. Assign values part - most of the optional parameters should have some kind
+1. Top comment: contains a list of all required and optional parameters that can be passed and will be used in this partial. This way it’s easier to see at a glance what options are available. 
+2. Assign values part: most of the optional parameters should have some kind
```
Comment (diana-lakatos) [inline marketplace_builder/views/pages/use-cases/e-commerce/using-shared-partials.liquid]: @chrisdanek This sentence is difficult to understand/unfinished. Could you please clarify?

File: marketplace_builder/views/pages/use-cases/e-commerce/using-shared-partials.liquid [hunk 4]
```diff
@@ -386,6 +386,6 @@ Congratulations! You’ve cleaned up the views and made it easier to reuse commo
 - extend default HTML5 validation by parsing `validation` rules provided by `form_builder`
 - use {% raw %}`{% log "{message}", type: "error" %}`{% endraw %} tag to mark an error anytime a require parameter is missing in the partial
 
-With these simple changes you are now ready to start building the admin panel in earnest. In the next part you’ll define all constants and enums for your application, so it’s easier than relying on magical strings.
+With these simple changes you are now ready to start building the admin panel in earnest. In the next part you’ll define all constants and enums for your application, so that it’s easier than relying on magical strings.
```
Comment (diana-lakatos) [inline marketplace_builder/views/pages/use-cases/e-commerce/using-shared-partials.liquid]: @chrisdanek Could you add some more explicit explanation for using constants and enums? Or we could just remove the explanation here, and explain it in the next tutorial.

**Review by diana-lakatos — CHANGES_REQUESTED** [review]

File: marketplace_builder/views/pages/use-cases/e-commerce/using-shared-partials.liquid (hunk 4 above)
Comment (chrisdanek) [reply marketplace_builder/views/pages/use-cases/e-commerce/using-shared-partials.liquid]: Next tutorial is all about this topic. I don’t know how to explain it in just one sentence

File: marketplace_builder/views/pages/use-cases/e-commerce/using-shared-partials.liquid (hunk 3 above)
Comment (chrisdanek) [reply marketplace_builder/views/pages/use-cases/e-commerce/using-shared-partials.liquid]: I get distracted easily... Finished the sentence now

**Review by lemingos — APPROVED** [review]

File: marketplace_builder/views/pages/use-cases/e-commerce/using-shared-partials.liquid (hunk 4 above)
Comment (diana-lakatos) [reply marketplace_builder/views/pages/use-cases/e-commerce/using-shared-partials.liquid]: I understand. Then I think we should remove the explanation here and leave it to the next topic.

**Review by diana-lakatos — APPROVED** [review]

**Pull request merged**